### PR TITLE
[MRG+1] Makes eventplot legend work

### PR DESCRIPTION
**Pull request by nepix32**
Fixes issue #7662

### CODE REVIEW

**Comment by tacaswell** [comment]
Can you also add a test to https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/tests/test_axes.py ?

something like 

```python

@cleanup
def test_eventplot_legend():
    # your code snippet from the bug report
```

**Comment by QuLogic** [comment]
Can you rebase and squash the first two commits together?

File: lib/matplotlib/tests/test_axes.py [hunk 1]
```diff
@@ -4871,3 +4871,8 @@ def test_scatter_color_masking():
     assert_array_equal(facecolors[1], np.array([0, 0, 0, 1]))
     assert_array_equal(linecolors[1], np.array([0, 0, 0, 1]))
     assert linewidths[1] == 3
+
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_axes.py]: One more blank line.

**Comment by QuLogic** [comment]
When you rebase, you need to force push (`git push --force`) and not pull again or you'll get two copies of your commits (new from you and old from GitHub).

**Comment by nepix32** [comment]
Think this all went horrible wrong (I hate now git to its guts), how to best clean up this mess:

* Delete this pull request
* Create a new one (with only the useful changes from above)
* Rebase if necessary

**Comment by QuLogic** [comment]
A new PR isn't necessary; just a rebase (and force push) will do. I can do that for you (you will need to reset your local copy.)

**Comment by nepix32** [comment]
I would say I have done enough damage already and ask you kindly to do the rebase and force/push.

**Comment by QuLogic** [comment]
BTW, your name on commits is sometimes `nepix32` and sometimes your email; what is the correct name you wish to use? (configure this with `git config --global user.name 'The name to use'`, but it only affects new commits.)

**Comment by nepix32** [comment]
nepix32 shall be ``user.name``, thanks for notice. Should be configured properly now.

**Comment by QuLogic** [comment]
I updated the PR; on your computer, 
```bash
$ git status
# Check for any changes you want to save (commit to a branch or stash them)
$ git fetch origin
$ git checkout nepix32-patch-1
$ git reset --hard origin/nepix32-patch-1
```

**Comment by nepix32** [comment]
Have done everything down to ``git reset ...`` and things look properly now locally. What is the **correct** way now to synchronize my local copy with the fork of matplotlib without breaking everything again? ``git push``?

**Comment by QuLogic** [comment]
I replaced your GitHub fork and the commands I gave above reset your local copy to match the fork. Everything should be in sync (as far as this PR/branch is concerned.)

**Comment by nepix32** [comment]
Ok, I see, was looking at the forks ``master``. In ``...patch...`` everything is fine. What do you recommend to reset the ``master`` in the fork to look exactly as the one in ``upstream``?

**Comment by QuLogic** [comment]
```bash
$ git checkout master
$ git reset --hard upstream/master
$ git push --force origin master
```
should get your local copy of `master` in sync with the version of here that your local copy believes to be the latest _and_ sync your fork with that.

In the future, (or now just to be sure you have the latest commit)
```
$ git checkout master
$ git fetch upstream
$ git merge upstream/master
$ git push origin master
```
should be a fast-forward merge that just moves `master` "up" to the latest commit.

**Comment by nepix32** [comment]
Hmmm... still have changes in ``origin/master`` I cannot get rid of. But anyway, many thanks for helping me to deal with ``git``!

**Comment by QuLogic** [comment]
Feel free to stop by our [Gitter room](https://gitter.im/matplotlib/matplotlib) for some help.

**Comment by NelleV** [comment]
Thanks for the patch!

**Pull request merged**